### PR TITLE
Workspace helper

### PR DESCRIPTION
**Pull request by rajivnathan**
Helper functions for space lister and tests

Depends on https://github.com/codeready-toolchain/api/pull/337

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]
Looks good but let's add a unit test for the helper.

File: pkg/proxy/workspace.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// server and unit/e2e tests only. It is not meant for use in any operator code.
+// See https://github.com/codeready-toolchain/api/pull/337 for more details.
+
+type WorkspaceOption func(*toolchainv1alpha1.Workspace)
+
+func NewWorkspace(name string, options ...WorkspaceOption) *toolchainv1alpha1.Workspace {
+	workspace := &toolchainv1alpha1.Workspace{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Workspace",
+			APIVersion: "toolchain.dev.openshift.com/v1alpha1",
```
Comment (mfrancisc) [inline pkg/proxy/workspace.go:18]:
```suggestion
			APIVersion: toolchainv1alpha1.LabelKeyPrefix+"v1alpha1",
```

**Review by mfrancisc — APPROVED** [review]
Looks good! just one small suggestion.

File: pkg/proxy/workspace.go (hunk 1 above)
Comment (rajivnathan) [reply pkg/proxy/workspace.go]: Right! Thanks!

**Comment by rajivnathan** [comment]
@alexeykazakov added unit tests in https://github.com/codeready-toolchain/toolchain-common/pull/290/commits/56df30fa01bb08152e0fdb90e5612519648cbd81

**Pull request merged**